lib/src/main/java/graphql/nadel/engine/transform/query/NadelQueryTransformer.kt

```diff
@@ -168,8 +168,14 @@ class NadelQueryTransformer private constructor(
     }
 
     private fun getUnderlyingTypeNames(objectTypeNames: Collection<String>): List<String> {
-        return objectTypeNames.map {
-            executionBlueprint.getUnderlyingTypeName(service, overallTypeName = it)
+        return if (executionContext.hints.sharedTypeRenames(service)) {
```

Comment:
this is the bit I missed the first time around. This is covered by the test where I added fragments that used renamed typenames
```graphql
  query {
    nodes {
      __typename
      ... on Issue {
        id
      }
      ... on JiraComment {
        id
      }
    }
  }
```
I didn't realise that this thing runs every time before we send the query to the underlying